src/pextlib1.0/Pextlib.c

```diff
@@ -135,6 +135,14 @@ static void ui_message(Tcl_Interp *interp, const char *severity, const char *for
     free(tclcmd);
 }
 
+__attribute__((format(printf, 3, 0)))
+static void ui_debug_x(Tcl_Interp *interp, unsigned int level, const char *format, va_list va) {
+    char cLevel[20];  // longer than necessary, but also safer
+
+    sprintf(cLevel, "debug%u", level);
```

Comment:
Please use `asprintf` as in the other functions or at least `snprintf`. Actually, I would just use a lookup table to map the level to the three possible strings.

Reply:
Fair enough, either would be preferable.

Reply:
Done, changed to use `snprintf`.